tests/test_helper.tcl

```diff
@@ -857,9 +857,24 @@ proc read_from_replication_stream {s} {
 }
 
 proc assert_replication_stream {s patterns} {
+    set errors 0
+    set values {}
     for {set j 0} {$j < [llength $patterns]} {incr j} {
-        assert_match [lindex $patterns $j] [read_from_replication_stream $s]
+        set pattern [lindex $patterns $j]
+        set value [read_from_replication_stream $s]
+        lappend values $value
+        if {![string match $pattern $value]} { incr errors }
```

Comment:
i suppose we stand a chance to hang here now (in case some expected reply is missing.
in the past it would have exited with error if the one after the missing one is mismatching what we're expecting, but now it can hang and we won't know anything...

please try to reproduce that and come up with a solution.

Reply:
yes, if some expected reply is missing, it will hang. but no for long, each reply will hang at most 1 second, i guess that should be enough
```
proc read_from_replication_stream {s} {
    fconfigure $s -blocking 0
    set attempt 0
    while {[gets $s count] == -1} {
        if {[incr attempt] == 10} return ""
        after 100
    }
    fconfigure $s -blocking 1
}
```

Reply:
ok, and when done will it print what it got from the stream till that point?
please make a quick experiment to see what happens in that case and make sure we have the info we need.

Reply:
yes, it will got a empty string for each expected key, now the new code will behave like this:
```
Testing unit/expire
[err]: GETEX propagate as to replica as PERSIST, DEL, or nothing in tests/unit/expire.tcl
Expected '{select 9} {set foo bar PXAT 1678282004920} {persist foo} {del foo} {} {} {}' to match '
           {select *}
           {set foo bar PXAT *}
           {persist foo}
           {del foo}
           {expected1}
           {expected2}
           {expected3}
        ' (context: type source line 872 file /root/redis/self/redis/tests/test_helper.tcl cmd {assert_match $patterns $values} proc ::assert_replication_stream level 1) 
[1/1 done]: unit/expire (3 seconds)

                   The End

Execution time of different units:
  3 seconds - unit/expire

!!! WARNING The following tests failed:

*** [err]: GETEX propagate as to replica as PERSIST, DEL, or nothing in tests/unit/expire.tcl
Expected '{select 9} {set foo bar PXAT 1678282004920} {persist foo} {del foo} {} {} {}' to match '
           {select *}
           {set foo bar PXAT *}
           {persist foo}
           {del foo}
           {expected1}
           {expected2}
           {expected3}
        ' (context: type source line 872 file /root/redis/self/redis/tests/test_helper.tcl cmd {assert_match $patterns $values} proc ::assert_replication_stream level 1) 
Cleanup: may take some time... OK
```